src/jrd/val.h

```diff
@@ -168,6 +168,13 @@ struct impure_value
 	void make_double(const double val);
 	void make_decimal128(const Firebird::Decimal128 val);
 	void make_decimal_fixed(const Firebird::Int128 val, const signed char scale);
+
+	template<class T>
+	VaryingString* getAllocateString(T requeuedLength, MemoryPool& pool) = delete; // Prevent dangerous length shrink
+	VaryingString* getAllocateString(USHORT requeuedLength, MemoryPool& pool);
```

Comment:
I'd call it just `getString()` for simplicity but I'm not going to insist.

Reply:
getAllocateString is not a name that sounds good.

Reply:
Usually pool comes first in parameter list.